File: tests/e2e/fixtures/code/sample.cs
// C# (legacy stream mode): classes, LINQ, string interpolation.
using System;
using System.Collections.Generic;
using System.Linq;

namespace NativeFileEditor
{
    public record Note(string Path, IReadOnlyList<string> Tags, long Size);

    public static class Scanner
    {
        private const long Limit = 5L * 1024 * 1024;

        public static IDictionary<string, List<Note>> GroupByTag(IEnumerable<Note> notes)
        {
            var big = notes.Where(n => n.Size > Limit).Select(n => n.Path).ToHashSet();
            return notes
                .Where(n => !big.Contains(n.Path))
                .SelectMany(n => n.Tags.Select(t => (t, n)))
                .GroupBy(p => p.t, p => p.n)
                .ToDictionary(g => g.Key, g => g.ToList());
        }

        public static void Main() => Console.WriteLine($"{DateTime.Now:O} ready");
    }
}

File: tests/e2e/fixtures/code/sample.rb
# Ruby (legacy stream mode): classes, blocks, symbols, interpolation.
require "pathname"

class Note
  LIMIT = 5 * 1024 * 1024
  attr_reader :path, :tags

  def initialize(path, tags = [])
    @path = Pathname(path)
    @tags = tags
  end

  def size = @path.size
  def large? = size > LIMIT
end

def group_by_tag(notes)
  notes.reject(&:large?).each_with_object(Hash.new { |h, k| h[k] = [] }) do |note, out|
    note.tags.each { |tag| out[tag] << note }
  end
end

puts "#{Time.now.iso8601} ready" if __FILE__ == $0
